js/vm/printjob.js

```diff
@@ -80,18 +83,42 @@ export default class PrintJob {
       price: this.totalPrice,
       printer: this.selectedPrinter,
     };
-    api.post('print', job, {
-      error: (xhr) => {
-        this.status = 'error';
-        if (xhr.status === 507)
-          this.errorText = JSON.parse(xhr.responseText).errors;
-      },
-    }).done(() => {
+
+    // See /api/print in odie-server. Everything's horrible.
+    $.cookie('print_data', JSON.stringify(job));
+    let stream = new EventSource(api.baseUrl + 'print');
+    stream.onerror = () => {
+      this.status = 'error';
+      this.errorText = '';
```

Comment:
A generic error message like "Verbindung zum Server verloren" would be nice here.